learning_journal_basic1/tests.py

```diff
@@ -1,29 +1,157 @@
-import unittest
+import pytest
+import transaction
 
 from pyramid import testing
 
+from ..models import (
+    MyModel,
+    get_engine,
+    get_session_factory,
+    get_tm_session,
+)
+from ..models.meta import Base
 
-class ViewTests(unittest.TestCase):
-    def setUp(self):
-        self.config = testing.setUp()
 
-    def tearDown(self):
+@pytest.fixture(scope="session")
+def sqlengine(request):
+    config = testing.setUp(settings={
+        'sqlalchemy.url': 'sqlite:///:memory:'
+    })
+    config.include("..models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
         testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+def test_model_gets_added(new_session):
+    assert len(new_session.query(MyModel).all()) == 0
+    model = MyModel(name="Bob", value=42)
+    new_session.add(model)
+    new_session.flush()
+    assert len(new_session.query(MyModel).all()) == 1
```

Comment:
What are other things about the model specifically that you can test to see if the model was added appropriately? What about adding multiple models?